src/ui/public/state_management/state.js

```diff
@@ -24,12 +24,16 @@ export default function StateProvider(Private, $rootScope, $location) {
         self.fetch();
       }),
 
-      // beginning of full route update, new app will be initialized before
-      // $routeChangeSuccess or $routeChangeError
       $rootScope.$on('$routeChangeSuccess', function () {
         if (self._persistAcrossApps) {
           self.fetch();
-        } else {
+        }
+      }),
+
+      // beginning of full route update, new app will be initialized before
+      // $routeChangeSuccess or $routeChangeError
+      $rootScope.$on('$routeChangeStart', function () {
```

Comment:
⚠️ super nit approaching ⚠️ 

can you swap these event handlers so that start if before success?